chore: merge tsconfig and use node16 module resolution

fryorcraken opened the pull request:
- No need to duplicate tsconfigs
- `node16` module resolution follows ESM rules, enabling `tsc` to catch any issue with imports + enables the usage of export maps.

fryorcraken commented on tsconfig.json:
```diff
@@ -0,0 +1,49 @@
+  "compilerOptions": {
+    "incremental": true,
+    "target": "es2020",
+    "moduleResolution": "node16",
```
This is the only value that has changed

danisharora099 reviewed (approved):
Looks good.

Why have we changed all file imports to a .js suffix manually and deleted the `fix-imports` script?

fryorcraken commented:
> Why have we changed all file imports to a .js suffix manually and deleted the fix-imports script?

Because now `tsc` fails if the extension is not present. Before it didn't care and the script was adding the extension in generated files (in `dist` folder).
